src/parse.rs

```diff
@@ -1406,3 +1569,130 @@ fn empty_array_expr() -> Box<Expr> {
     span: DUMMY_SP,
   }))
 }
+
+fn get_expr_span<'a>(expr: &'a Expr) -> &'a Span {
+  match expr {
+    Expr::Array(e) => &e.span,
+    Expr::Arrow(e) => &e.span,
+    Expr::Assign(e) => &e.span,
+    Expr::Await(e) => &e.span,
+    Expr::Bin(e) => &e.span,
+    Expr::Call(e) => &e.span,
+    Expr::Class(e) => &e.class.span,
+    Expr::Cond(e) => &e.span,
+    Expr::Fn(e) => &e.function.span,
+    Expr::Ident(e) => &e.span,
+    Expr::Invalid(e) => &e.span,
+    Expr::This(e) => &e.span,
+    Expr::Object(e) => &e.span,
+    Expr::Unary(e) => &e.span,
+    Expr::Update(e) => &e.span,
+    Expr::Member(e) => &e.span,
+    Expr::SuperProp(e) => &e.span,
+    Expr::New(e) => &e.span,
+    Expr::Seq(e) => &e.span,
+    Expr::Lit(e) => get_lit_span(e),
+    Expr::Tpl(e) => &e.span,
+    Expr::TaggedTpl(e) => &e.span,
+    Expr::Yield(e) => &e.span,
+    Expr::MetaProp(e) => &e.span,
+    Expr::Paren(e) => &e.span,
+    Expr::JSXMember(e) => get_jsx_object_span(&e.obj), // TODO: combine spans? this is wrong, why don't these nodes have spans?
+    Expr::JSXNamespacedName(e) => &e.name.span, // TODO: combine spans? this is wrong, why don't these nodes have spans?
+    Expr::JSXEmpty(e) => &e.span,
+    Expr::JSXElement(e) => &e.span,
+    Expr::JSXFragment(e) => &e.span,
+    Expr::TsTypeAssertion(e) => &e.span,
+    Expr::TsConstAssertion(e) => &e.span,
+    Expr::TsNonNull(e) => &e.span,
+    Expr::TsAs(e) => &e.span,
+    Expr::TsInstantiation(e) => &e.span,
+    Expr::PrivateName(e) => &e.span,
+    Expr::OptChain(e) => &e.span,
+  }
+}
```

Comment:
Can this be simplified with a default? Or no because rust isn't structurally typed and it wouldn't narrow to have `span` on the last value?

Reply:
Not sure what you mean?